### PR TITLE
[R2] demo worker example fix

### PR DESCRIPTION
**Pull request by deadlypants1973**
closes #5516

### CODE REVIEW

**Review by ranbel — APPROVED** [review]

File: content/r2/examples/demo-worker.md [hunk 1]
```diff
@@ -67,9 +50,8 @@ export default {
       }
 
       if (request.method === 'GET') {
-        const range = parseRange(request.headers.get('range'))
         const object = await env.MY_BUCKET.get(objectName, {
-          range,
+          range: request.headers,
```
Comment (KianNH) [inline content/r2/examples/demo-worker.md]: This should be sound, but I can't get it to work in any variety without throwing an internal error. 

Spoken to Harshal who is going to see if he can figure out why it's throwing that.

File: content/r2/examples/demo-worker.md [hunk 2]
```diff
@@ -80,10 +62,10 @@ export default {
         const headers = new Headers()
         object.writeHttpMetadata(headers)
         headers.set('etag', object.httpEtag)
-        if (range) {
-          headers.set("content-range", `bytes ${range.offset}-${range.end}/${object.size}`)
+        if (object.range) {
```
Comment (KianNH) [inline content/r2/examples/demo-worker.md]: `object.range` will exist even on a non-ranged request so we'll likely have to continue checking for the existence of a `range` header on the request.

File: content/r2/examples/demo-worker.md [hunk 3]
```diff
@@ -80,10 +62,10 @@ export default {
...
+          headers.set("content-range", `bytes ${object.range.offset}-${object.range.end}/${object.size}`)
         }
-        const status = object.body ? (range ? 206 : 200) : 304
+        const status = object.body ? (object.range ? 206 : 200) : 304
```
Comment (KianNH) [inline content/r2/examples/demo-worker.md]: Same as above - but one question is on the semantics of 206.

Do we always return 206 in response to a ranged request, even if we return the full content? Will need to do some RFC/MDN reading.

Comment (vlovich) [reply content/r2/examples/demo-worker.md]: From my reading, yes. MDN indicates 206 is returned always even if you end up satisfying the entire file.

File: content/r2/examples/demo-worker.md (hunk 1 above)
Comment (Frederik-Baetens) [reply content/r2/examples/demo-worker.md]: This should be fixed now. Let me know if you need anything else from me to unblock merging this PR.

**Pull request merged**